### PR TITLE
Fix the `examples/app_dag` App

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes the `examples/app_dag` App

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: tests/tests_app_examples/test_app_dag.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+from lightning_app.testing.testing import run_app_in_cloud
+
+
+@pytest.mark.cloud
+def test_app_dag_example_cloud() -> None:
+    with run_app_in_cloud(os.path.join(_PROJECT_ROOT, "examples/app_dag")) as (_, _, fetch_logs, _):
+
+        launch_log, finish_log = False, False
+        while not launch_log and not finish_log:
```
Comment (hhsecond) [inline tests/tests_app_examples/test_app_dag.py:15]:
```suggestion
        while not (launch_log and finish_log):
```

Comment (hhsecond) [reply tests/tests_app_examples/test_app_dag.py:15]: ?

File: examples/app_dag/app.py [hunk 2]
```diff
@@ -80,12 +80,10 @@ def __init__(self, models_paths):
     def run(self):
         # Step 1 and 2: Download and process the data.
         self.data_collector.run()
-        self.data_collector.stop()  # Stop the data_collector to reduce cost
```
Comment (hhsecond) [inline examples/app_dag/app.py:83]: what was wrong with this?

Comment (kaushikb11) [reply examples/app_dag/app.py:83]: As the work was stopped.

`self.data_collector.df_data` was None.

**Review by hhsecond — COMMENTED** [review]
We shouldn't add this e2e to our regular e2e suite since this example doesn't really test any functionality of our system specifically. But perhaps to another job, like what Mansy did for making sure the app is not broken

**Comment by kaushikb11** [comment]
@hhsecond @manskx 

Makes sense. Could you guide where would be the right place to add the test?

**Comment by manskx** [comment]
> 

Yes, I think it's overkill to run all the cloud apps on each PR, we can have them on a corn job on something. @kaushikb11 let's sync to plan this.

File: examples/app_dag/app.py (hunk 2 above)
Comment (manskx) [reply examples/app_dag/app.py:83]: I don't have enough context TBH

Comment (awaelchli) [reply examples/app_dag/app.py:83]: There are other stop calls in the loop below

Comment (hhsecond) [reply examples/app_dag/app.py:83]: But that's only for `ModelWork` no?

**Review by awaelchli — APPROVED** [review]

File: examples/app_dag/app.py (hunk 2 above)
Comment (tchaton) [reply examples/app_dag/app.py:83]: Normally, they were blocking, so it should have worked.

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by Borda** [comment]
@kaushikb11, how is it going here? close to merge... :otter:

**Review by Borda — APPROVED** [review]

**Comment by justusschock** [comment]
@tchaton If you don't feel this is relevant anymore feel free to close it :)

**Pull request merged**